reference-architecture/aws-ansible/playbooks/roles/s3-registry-bucket/tasks/main.yaml

```diff
@@ -1,6 +1,10 @@
 ---
 - name: set the name of the bucket
-  set_fact: s3_bucket="{{ ansible_date_time.epoch }}-openshift-docker-registry"
+  set_fact: s3_bucket="{{ ansible_date_time.epoch }}-openshift-docker-registry-{{ stack_name}}"
```

Comment:
@cooktheryan seems a bit fragile, that said if you do take this approach, I would suggest adding tags to the s3 bucket and adding some checks for a s3 bucket with the proper tags to conditionalize the creation.

For the nested_stacks, I'm planning on supporting env_id and cluster_id and requiring that combinations of those be globally unique, could probably leverage something similar for the s3 bucket...  might want an early "pre-req" check to validate that the bucket doesn't exist or exists in the right region, though.